partition_table_engine/src/partition.rs

```diff
@@ -84,6 +99,86 @@ impl PartitionTableImpl {
             table: format_sub_partition_table_name(&self.table_data.table_name, &partition_name),
         }
     }
+
+    async fn write_single_row_group(
+        &self,
+        partition_id: usize,
+        row_group: RowGroup,
+    ) -> Result<usize> {
+        let sub_table_ident = self.get_sub_table_ident(partition_id);
+
+        let request = RemoteWriteRequest {
+            table: sub_table_ident,
+            write_request: WriteRequest { row_group },
+        };
+
+        self.remote_engine
+            .write(request)
+            .await
+            .box_err()
+            .with_context(|| WriteBatch {
+                tables: vec![self.table_data.table_name.clone()],
+            })
+    }
+
+    async fn write_partitioned_row_groups(
+        &self,
+        schema: Schema,
+        partitioned_rows: PartitionedRowsIter,
+    ) -> Result<usize> {
+        let mut split_rows = HashMap::new();
+        for PartitionedRow { partition_id, row } in partitioned_rows {
+            split_rows
+                .entry(partition_id)
+                .or_insert_with(Vec::new)
+                .push(row);
+        }
```

Comment:
How about move here into `location_partitions_for_write`, and just return final `HashMap<partition, RowGroup>`?